wasm/src/lib.rs

```diff
@@ -0,0 +1,148 @@
+use sha2::{Digest, Sha256};
+
+cfg_if::cfg_if! {
+	if #[cfg(consolelog)] {
+		use web_sys::console;
+
+		fn console_log(info: &str) {
+			console::log_1(&info.into());
+		}
+	} else {
+		fn console_log(_info: &str) {}
+	}
+}
+
+#[wasm_bindgen]
+pub fn setup() {
+	console_log("@solana/wasm Initialized. WASM ready for calls.");
+	utils::set_panic_hook();
+}
+
+#[wasm_bindgen]
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct Pair{
+	public_key: Vec<u8>,
+	secret_key: Vec<u8>
+}
+
+/// Generate key pair for use with ED25519 from ed25519_dalek
+///
+/// * returned struct has two fields public and secret.
+#[wasm_bindgen(js_name = generateKeyPair)]
+pub fn generate_keypair() -> Result<JsValue, JsValue> {
+	// using get random directly instead of OsRng since it's easier to integrate with WASM
+	let mut seed = [0u8; 32];
+	getrandom::getrandom(&mut seed).unwrap();
+	let secret = SecretKey::from_bytes(&seed).unwrap();
+	let public: PublicKey = (&secret).into();
+	let keypair = Keypair{secret, public};
+
+	// use all 64bit for secret key to maintain compatibility with tweetnacl
+	let secret_key = keypair.to_bytes().to_vec();
+	let public_key = keypair.public.as_bytes().to_vec();
+
+	let result = Pair{ public_key, secret_key};
+	Ok(JsValue::from_serde(&result).unwrap())
+}
+
+/// Create key pair from secret key for use with ED25519 from ed25519_dalek
+/// * secretkey: UIntArray with 64 elements
+///
+/// * returned struct has two fields public and secret.
+#[wasm_bindgen(js_name = keyPairFromSecretKey)]
+pub fn from_secret_key(secretkey: &[u8]) -> Result<JsValue, JsValue> {
+	// use 64bit for secret key to maintain compatibility with tweetnacl
+	let keypair = Keypair::from_bytes(&secretkey).unwrap();
+	let secret_key = keypair.to_bytes().to_vec();
+	let public_key = keypair.public.as_bytes().to_vec();
+
+	let result = Pair{ public_key, secret_key};
+	return Ok(JsValue::from_serde(&result).unwrap());
+}
+
+/// Create key pair from secret key for use with ED25519 from ed25519_dalek
+/// * seed: secret key UIntArray with 32 elements
+///
+/// * returned struct has two fields public and secret.
+#[wasm_bindgen(js_name = keyPairFromSeed)]
+pub fn keypair_from_seed(seed: &[u8]) -> Result<JsValue, JsValue> {
+	// use 64bit for secret key to maintain compatibility with tweetnacl
+	let secret = SecretKey::from_bytes(&seed).unwrap();
+	let public: PublicKey = (&secret).into();
+	let keypair = Keypair{secret, public};
+	let public_key = public.as_bytes().to_vec();
+	let secret_key = keypair.to_bytes().to_vec();
+
+	let result = Pair{ public_key, secret_key};
+	return Ok(JsValue::from_serde(&result).unwrap());
+}
+
+/// Sign a message using ED25519 from ed25519_dalek
+/// * pubkey: UIntArray with 32 elements
+/// * private: UIntArray with 32 or 64 elements
+/// * message: Arbitrary length UIntArray
+///
+/// * returned vector is the signature consisting of 64 bytes.
+#[wasm_bindgen(js_name = signED25519)]
```

Comment:
nit: Ed25519 is not usually written in all caps. Also, since we only support `ed25519` signatures, we don't need to specify the scheme in the method name, just `sign` is sufficient